sql/bloom_filters.h

```diff
@@ -177,9 +184,153 @@ struct PatternedSimdBloomFilter
      basically, unnoticeable, well below the noise level */
 #endif
 
+#ifdef NEON_IMPLEMENTATION
+  uint64x2_t CalcHash(uint64x2_t vecData)
+  {
+    static constexpr uint64_t prime_mx2= 0x9FB21C651E98DF25ULL;
+    static constexpr uint64_t bitflip= 0xC73AB174C5ECD5A2ULL;
+    uint64x2_t step1= veorq_u64(vecData, vdupq_n_u64(bitflip));
+    uint64x2_t step2= veorq_u64(vshrq_n_u64(step1, 48), vshlq_n_u64(step1, 16));
+    uint64x2_t step3= veorq_u64(vshrq_n_u64(step1, 24), vshlq_n_u64(step1, 40));
+    uint64x2_t step4= veorq_u64(step1, veorq_u64(step2, step3));
+    uint64x2_t step5;
+    step5= vsetq_lane_u64(vgetq_lane_u64(step4, 0) * prime_mx2, step4, 0);
+    step5= vsetq_lane_u64(vgetq_lane_u64(step4, 1) * prime_mx2, step5, 1);
```

Comment:
strange. it doesn't have 64-bit vector multiplication at all?

Reply:
Correct, I couldn't find any. And I don't seem to be alone, google for "neon multiply 64".